client/state/plugins/marketplace/reducer.ts

```diff
@@ -40,19 +50,66 @@ export function purchaseFlow(
 			action = action as ISetPluginToBeInstalledAction;
 			return {
 				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
 				pluginSlugToBeInstalled: action.pluginSlugToBeInstalled,
 			};
 		case MARKETPLACE_PLUGIN_INSTALLED_ON_PURCHASE:
 			action = action as ISetPluginInstalledDuringPurchaseFlag;
 			return {
 				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
 				isPluginInstalledDuringPurchase: action.isPluginInstalledDuringPurchase,
 			};
+		case THEME_TRANSFER_INITIATE_REQUEST:
+			return {
+				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN,
+			};
+		case MARKETPLACE_SITE_TRANSFER_STATE_CHANGE: {
+			const { isPluginInstalledAlongWithTransfer } = state;
+			if (
+				isPluginInstalledAlongWithTransfer &&
+				action.state === MARKETPLACE_ASYNC_PROCESS_STATUS.COMPLETED
+			) {
+				/**
+				 * If the site transfer happened together with a plugin installation
+				 * We mark the plugin installation as completed as well
+				 */
+				return {
+					...state,
+					siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.COMPLETED,
+					reasonForSiteTransferStatus: 'Site transfer and plugin install done together',
+					pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.COMPLETED,
+					reasonForPluginInstallationStatus: 'Site transfer and plugin install done together',
+				};
+			}
+			return {
+				...state,
+				siteTransferStatus: action.state,
+				reasonForSiteTransferStatus: action.reason,
+			};
+		}
+		case MARKETPLACE_SITE_TRANSFER_PLUGIN_INSTALL:
+			return {
+				...state,
+				siteTransferStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS,
+				pluginInstallationStatus: MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS,
+				isPluginInstalledAlongWithTransfer: true,
+			};
+		case MARKETPLACE_PLUGIN_INSTALLATION_STATE_CHANGE:
+			return {
+				...state,
+				pluginInstallationStatus: action.state,
+				reasonForPluginInstallationStatus: action.reason,
+			};
 		default:
 			return state;
 	}
 }
 
 export default combineReducers( {
-	purchaseFlow,
+	purchaseFlow: withPersistence( purchaseFlow ),
```

Comment:
What's the idea of using `withPersistence` here? Do we mean to persist the entire tree instead?

Reply:
Ah didn't give it much thought right now since we are only looking at the purchase flow right now. Do I wrap this around combined reducers to make the whole slice persisted? Not entirely sure how this works even though I tried to dig deep a bit :D. Any resources to understand how persistence works here?

Reply:
It's just fine to wrap only an inner part of the reducer 😉 Just wanted to ensure that is the intention.

And as to documentation, this is a good read that also offers some alternatives based on what you are trying to achieve: https://github.com/Automattic/wp-calypso/blob/update/my-home-views-no-redux/docs/data-persistence.md